### PR TITLE
tests for employee

### PR DESCRIPTION
**Pull request by Bazhanel**

### CODE REVIEW

File: src/main/java/com/danit/services/employee/CompanyService.java [hunk 1]
```diff
@@ -18,4 +18,6 @@
 
   boolean companyExists(long id);
 
+  int getCompanyQuant();
```
Comment (Thealoner) [inline src/main/java/com/danit/services/employee/CompanyService.java]: I'd recommend to use either complete words (e.g. 'Quantity') or popular abbreviations ('Qty').

File: src/test/java/com/danit/controllers/employee/CompanyControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  }
+
+  @Test
+  public void createCompany() throws Exception {
+    int currentQuant = companyService.getCompanyQuant();
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+    this.mockMvc.perform(post(url).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"sname\": \"Energym\",\n"
```
Comment (Thealoner) [inline src/test/java/com/danit/controllers/employee/CompanyControllerTest.java]: It is unclear to me what 'sname' is. Should we use a more elaborate term?

File: src/test/java/com/danit/controllers/employee/GymControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            + "  }"))
+        .andExpect(status().isOk());
+
+    mockMvc.perform(get(url).headers(header))
+        .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$", hasSize(currentQuant + 1)));
+  }
+
+  @Test
+  public void expect500WhenNoDataFoundService() throws Exception {
```
Comment (Thealoner) [inline src/test/java/com/danit/controllers/employee/GymControllerTest.java]: expect**404**WhenNoDataFoundService?

File: src/test/java/com/danit/controllers/employee/GymControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+
+    mockMvc.perform(get(url+"/0").headers(header))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  public void expect500WhenDeleteNonexistentService() throws Exception {
```
Comment (Thealoner) [inline src/test/java/com/danit/controllers/employee/GymControllerTest.java]: Ideally DELETE non-existing object should return 404 too.

File: src/test/java/com/danit/controllers/employee/TrainingTypeControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            + "  }"))
+        .andExpect(status().isOk());
+
+    mockMvc.perform(get(url).headers(header))
+        .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$", hasSize(currentQuant + 1)));
+  }
+
+  @Test
+  public void expectIsNotFoundWhenNoDataFoundService() throws Exception {
```
Comment (Thealoner) [inline src/test/java/com/danit/controllers/employee/TrainingTypeControllerTest.java]: expectIs404WhenNoDataFoundTrainingType?

File: src/test/java/com/danit/controllers/employee/TrainingTypeControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+
+    mockMvc.perform(get(url+"/0").headers(header))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  public void expect500WhenDeleteNonexistentService() throws Exception {
```
Comment (Thealoner) [inline src/test/java/com/danit/controllers/employee/TrainingTypeControllerTest.java]: expect500WhenDeleteNonexistentTrainingType
or could be 
expect404WhenDeleteNonexistentTrainingType
if you agree on returning 404 for deleting non-existing objects

File: src/test/java/com/danit/controllers/employee/GymControllerTest.java (hunk 4 above)
Comment (Bazhanel) [reply src/test/java/com/danit/controllers/employee/GymControllerTest.java]: mockMvc return 500

File: src/test/java/com/danit/controllers/employee/TrainingTypeControllerTest.java (hunk 6 above)
Comment (Bazhanel) [reply src/test/java/com/danit/controllers/employee/TrainingTypeControllerTest.java]: delete returns 500 if object doesn't exists

**Pull request merged**